src/txkube/_network.py

```diff
@@ -341,8 +371,38 @@ class _NetworkKubernetes(object):
         default=attr.Factory(lambda: Agent(namedAny("twisted.internet.reactor"))),
     )
 
+    def versioned_client(self):
+        """
+        Ask the server for its version and then create a client using a matching
+        model.
+        """
+        client = self.client()
+        d = client.version()
+        d.addCallback(self._version_to_client, client)
+        return d
+
+
+    _versions = {
+        (u"1", u"5"): v1_5_model,
+        (u"1", u"6"): v1_6_model,
+        (u"1", u"7"): v1_7_model,
+    }
+
+    def _version_to_client(self, version, client):
+        try:
+            model = self._versions[version.major, version.minor]
+        except KeyError:
+            raise ValueError(
+                "Unsupported Kubernetes version: {}.{}".format(
```

Comment:
❓ It would have been clearer if the exception had included the full version.Info repr.